src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: Array<string | number>,
+  key: string | number,
+): GroupsMap {
+  const result: GroupsMap = {};
+
+  for (const item of items) {
+    if (result.hasOwnProperty(item[key])) {
+      result[item[key]].push(item);
+    } else {
+      result[item[key]] = [item];
+    }
```

Comment:
The items in the array are expected to be objects since you're using `item[key]` to access their properties. However, the `items` parameter is typed as an array of `string | number`, which doesn't have properties. You should revise the type of `items` to reflect that it should be an array of objects.